### DISCUSSION TITLE
`cache add` multiple packages

### DISCUSSION THREAD

**Discussion by mjsir911** [npm/feedback#322] [CLI] [answered]
I would like to be able to add multiple tarballs to the cache with `npm cache add`.

Currently, while adding a few packages in succession with `cache add` is fine, you really notice the slowdowns with hundreds of packages.

The documentation for [npm-cache](https://docs.npmjs.com/cli/v6/commands/npm-cache) makes it seem like this suggestion is backwards compatible, just accepting `npm cache add a b c` instead of `npm cache add a; npm cache add b; npm cache add c`.

Unfortunately, theres some undocumented behaviour where arg 1 is the package name and arg 2 is the version, so this suggestion breaks that.

This feature request is part of my cache adventures outlined in npm/cli#2976

My current workaround for these slowdowns are to re-implement `npm cache` in bash, much faster & supports multiple integrity types until npm/cacache#46 is good to go.

The fix for the isn't too difficult, and I'm patching my own `npm` to do this, but the backwards-incompatibility complicates things

**Comment by MylesBorins** [comment, marked as answer]
@mjsir911 the @npm/cli team are the ultimate authority and decision makers about changes to the CLI and associated packages. It seems like you are already engaging in their process so I'm not sure there is much that can be done through this repo.

**Reply by mjsir911** [reply]
@MylesBorins, my work with the CLI has been bugfixes while this discussion is talking about a feature, and a backwards-incompatible feature at that. 

I'm not sure how to proceed with this feature request if not here since the [npm/cli feature requests](https://github.com/npm/cli/issues/new/choose) point to this repository

**Reply by MylesBorins** [reply]
High level feature requests are triaged here but get forwarded to https://github.com/npm/rfcs to make a formal proposal.

As you already have a PR open on https://github.com/npm/pacote/pull/74 it seemed like there wasn't a need to point you towards that process.  That said that would be the other possible step to take other than continuing with the existing PRs you have open. Apologies that the process is a bit opaque
